pkg/loadbalancer/experimental/lbmaps.go

```diff
@@ -60,6 +67,100 @@ func newLBMapsConfig(dcfg *option.DaemonConfig) (cfg LBMapsConfig) {
 	return
 }
 
+func newLBMapsCommand(m LBMaps) hive.ScriptCmdOut {
+	if f, ok := m.(*FaultyLBMaps); ok {
+		m = f.impl
+	}
+
+	var snapshot *mapSnapshots
+	errUsage := fmt.Errorf("%w: expected 'dump', 'snapshot' or 'restore'", script.ErrUsage)
+	return hive.NewScriptCmd(
+		"lb-maps",
+		script.Command(
+			script.CmdUsage{
+				Summary: "interact with load-balancing BPF maps",
+				Args:    "cmd (args...)",
+				Detail: []string{
+					"Supported commands:",
+					"  cmp <file>:   Compare the maps against a file. Retries.",
+					"  dump <file>:  Dump maps to stdout or file if specified",
+					"  snapshot:     Snapshot the maps",
+					"  restore:      Restore from snapshot",
+				},
+			},
+			func(s *script.State, args ...string) (script.WaitFunc, error) {
+				if len(args) < 1 {
+					return nil, errUsage
+				}
+
+				switch args[0] {
+				case "dump":
+					return func(s *script.State) (stdout string, stderr string, err error) {
+						out := DumpLBMaps(
+							m,
+							loadbalancer.L3n4Addr{},
+							false,
+							nil,
+						)
+						data := strings.Join(out, "\n") + "\n"
+						if len(args) == 2 {
+							err = os.WriteFile(s.Path(args[1]), []byte(data), 0644)
+						} else {
+							stdout = data
+						}
+						return
+					}, nil
+				case "cmp":
+					if len(args) != 2 {
+						return nil, errUsage
+					}
+					b, err := os.ReadFile(s.Path(args[1]))
+					if err != nil {
+						return nil, fmt.Errorf("read %q: %w", args[1], err)
+					}
+
+					expected := difflib.SplitLines(string(b))
+
+					var diff string
+					for range 10 {
```

Comment:
I'm not quite following why we need this for loop?

Reply:
Reconciliation is async so we don’t know exactly when it’s done and hence need to try few times. Same as with ”db cmp”. I might think about adding some support to reconciler to allow some sort of syncing for tests.

Reply:
Oh, I see now, we exit early if the len(diff) == 0. 

Wondering if it would make sense to extract the maxRetries and sleep as const? 
Also, I assume this pattern will be fairly common in the future, should we have some generic function for retries?

Reply:
Yeah probably later I'll think about adding some sort of "eventually" thing that wraps the commands. It's not entirely trivial so didn't look into it yet.

Reply:
Here we go: https://github.com/cilium/hive/pull/22